Feature/product list search rest api customer filter

gengjozsef opened the pull request.

julianzimmermann commented on bundles/product-locale-restriction-storage/src/FondOfOryx/Zed/ProductLocaleRestrictionStorage/ProductLocaleRestrictionStorageConfig.php:
```diff
@@ -0,0 +1,34 @@
+     */
+    public function isSendingToQueue(): bool
+    {
+        return true;
```
For what you need this, since its always true?

daniel-rose replied:
For overwriting this by pyz...

daniel-rose committed:
Allow running unit tests in monorepo (#488)

- add require workflows in split repos
- add monorepo env in split repos
- fix unit tests

daniel-rose committed:
Add jellyfish sales order company unit address bundle (#489)

- add plugin to expand jellyfish order with company unit address id
- add unit tests
- add entry to dandelion.json for creating split repo
- ...

daniel-rose committed:
Add jellyfish sales order customer address bundle (#490)

- add plugin to expand jellyfish order address with customer address id
- add unit tests
- add entry to dandelion.json for creating split repo
- ...

daniel-rose committed:
Add jellyfish sales order vat id bundle (#491)

- add plugin to expand jellyfish order with vat id
- apply vat id from sales order address to jellyfish order address
- add unit tests
- add entry to dandelion.json for creating split repo
- ...

daniel-rose committed:
Use from and to array in mappings (#492)

daniel-rose committed:
Add jellyfish sales order company user bundle (#494)

- add transfer definition to map company user reference correctly
- add unit tests
- add entry to dandelion.json for creating split repo
- ...

daniel-rose committed:
Set first and last name to null by default (#493)

daniel-rose committed:
Release jellyfish sales order bundle (#495)

- add phpdoc fixes (use correct return values) 🪲
- use from- and to-array in address- and order-mapper ⚙️
- use github actions for split repo 🚀
- increase minor version 🔢
- add return type to allow newest spryker log bundle 📝
- ...

daniel-rose committed:
Fix typo in composer json (#499)

- affected bundle: jellyfish-sales-order-vat-id
- ...

daniel-rose committed:
Make bundle independent to symfony stuff (#500)

- remove dependency symfony/http-foundation
- use fix values instead of constants form symfony/http-foundation
- prepare unit test
- increase minor version in dandelion.json
- ...

daniel-rose committed:
Remove duplicate foreign keys (#501)

daniel-rose committed:
Remove first- and last-name from jellyfish sales order bundle (#502)

DerCalli committed:
Payment method restriction by countries

* Skeleton

julianhyatt committed:
:sparkles: add company oms mail connector (#504)

:sparkles: add company oms mail connector

DerCalli committed:
🧩  Package provides plugins for payment-restrictions based on address-data

julianhyatt committed:
:children_crossing: use fallback for missing locale (#507)

DerCalli committed:
remove payment-address-restriction from dandelion.json (#508)

DerCalli committed:
Feature: payment address restrictions

* add plugin if billing address-equals shipping-address
* add plugin check if payment method is country ristrected
* add package to dandelion.json
* refactor, improve readability

daniel-rose committed:
:package: Initialize country oms mail connector bundle (#509)

- add contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- add required transfer definition
- ...

daniel-rose committed:
:package: Implement contracts of country oms mail connector bundle (#510)

- expand billing address with country (iso2code, iso3code, ...)
- expand shipping address on item level with country (iso2code, iso3code, ...)
- expand shipping address on order level with country (iso2code, iso3code, ...)
- add new plugin for expanding
- add unit tests
- ...

daniel-rose committed:
Release country oms mail connector bundle (#511)

julianhyatt committed:
:sparkles: prepare release for erp delivery note packages (#512)

julianhyatt committed:
:white_check_mark: fix to pass ci (#513)

julianhyatt committed:
:white_check_mark: fix to pass ci (#514)

julianhyatt committed:
:white_check_mark: fix to pass ci (#516)

DerCalli committed:
payment total amount restriction (#515)

* skeleton

DerCalli committed:
payment total amount restriction

* business logic + unit-tests

julianhyatt committed:
:white_check_mark: fix to pass ci (#519)

DerCalli committed:
skeleton (#520)

DerCalli committed:
payment total amount restriction

* add payment-total-amount-restriction:1.0.0 to dandelion.json

DerCalli committed:
💵  payment product restriction (#521)

* business logic + unit tests
* allow exclude multiple sku patterns per payment method

DerCalli committed:
add basic usage for all modules to readme

- add basic usage to readme for payment-address-restriction
- add basic usage to readme for payment-product-restriction
- add basic usage to readme for payment-total-amount-restriction
- add payment-product-restriction version

daniel-rose committed:
✨ Add fulltext search to company business unit address search rest api bundle (#523)

- add query to transfer definition
- make fulltext search fields configurable
- prepare unit tests
- increase minor version in dandelion.json
- ...

julianhyatt committed:
:bug: add fallback for company locale (#524)

* :bug: Add locale fallback if no locale is present

julianhyatt committed:
:bug: fix company locale (#525)

julianhyatt committed:
:bookmark: prepare release (#526)

julianhyatt committed:
:sparkles: Add GiftCardApi for searching gift cards by code (#527)

* :sparkles: Add GiftCardApi Bundle

gengjozsef committed:
implement CompanyUsersRestApiExtension (#528)

* add CompanyUsersRestApiExtension

* phpcs fixes

* phpcs fix

* add autoload

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

DerCalli committed:
🧩  payment product restriction filter (#529)

* remove sku-pattern based payment-filter, replaced by product-attribute based filter, adjusted unit tests
* update readme
* payment-product-restriction version raised to 1.0.1

DerCalli committed:
💵  payone credit memo

* move listening payment methods to config + unit-tests
* unit-tests
* Version of fond-of-oryx/payone-credit-memo upgraded to 1.0.1

DerCalli committed:
Update credit-memo

* create new transfers instead using from/to-array

julianhyatt committed:
:sparkles: Adds payone credit memo extension (#532)

:sparkles: Provides new interface for modifying pre refund plugins

julianhyatt committed:
:sparkles: Introduce new interface (#533)

:sparkles: Add new interface for mapping more info to credit memo transfers
:green_heart: Adds ci stuff

julianhyatt committed:
:sparkles: Some improvements (#534)

:sparkles: Add functionality to run expander plugins for entity to transfer mapper
:heavy_plus_sign: Require CreditMemoExtension ^1.1.0
:green_heart: Add ci stuff
:recycle: some refactoring to pass tests
:heavy_plus_sign: add missing dependencies

julianhyatt committed:
:bookmark: prepare for release CreditMemo 1.1.0 (#535)

julianhyatt committed:
:sparkles: Feature/credit memo gift card connector (#536)

* :sparkles: Add CreditMemoGiftCardConnector bundle

julianhyatt committed:
Feature/payone credit memo gift card connector (#537)

* :sparkles: Add PayoneCreditMemoGiftCardConnector bundle
Provides plugin for recalculation the payone refund amount

julianhyatt committed:
:bookmark: prepare for release (#538)

julianhyatt committed:
✨  Feature/payone credit memo (#539)

:sparkles: Implements pre refund plugin execution
:recycle: Moved business logic from command plugin to business layer
:arrow_up: Upgrade PayoneCreditMemoExtension to 1.1.0 to use new plugin interface
:green_heart: Add CI stuff
:white_check_mark: fix to pass ci

daniel-rose committed:
💄 Use newest phpcs ruleset from spryker (#541)

* 💄 Use newest phpcs ruleset from spryker
* Fix phpstan errors

DerCalli committed:
feature/payment product restriction: remove mapping (#540)

* remove sku-pattern based payment-filter, replaced by product-attribute based filter
* adjusted unit tests
* update README.md

gengjozsef committed:
extend postCreate with new argument type RestCompanyUsersRequestAttributesTransfer (#542)

* add CompanyUsersRestApiExtension

* phpcs fixes

* phpcs fix

* add autoload

* add RestCompanyUsersRequestAttributesTransfer as second parameter to the method postCreate

* phpcs fix

* phpcs fix

* fix phpstan: add transfer.xml

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
implement company-user-company-role-connector (#543)

* implement saveCompanyUserCompanyRole

* add Unit Tests

* add autoloads

* extend Transfers

* add Plugin Test

* add CompanyUsersRestApiExtension autoload in composer.json

* add Facade, BusinessFactory and Writer Tests

* code cleanup

* phpcs fixes

* phpcs fix

* phpcs fix

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

DerCalli committed:
fix typo in composer.json (#544)

version of fond-of-oryx/payment-address-restriction upgraded to 1.0.1

DerCalli committed:
payment product restriction (#545)

* product attribut is a separated list instead array
* use beta tags for invoice modules

daniel-rose committed:
🧩 Add erp customer bundle (#546)

- add external reference to customer table
- add entry to dandelion.json for creating split repo
- ...

julianhyatt committed:
💄 Split fond-of-spryler/log into connector packages

- add log-aws-cloud-watch-connector
- add log-slack-connector
- use newest log package
- ...

daniel-rose committed:
💄 Rename log.io package (#548)

- use the same naming structure from the other log packages
- add github action workflow
- ...

gengjozsef committed:
add company-user-mail-connector (#547)

* implement interfaces and initial files

* phpcs fix

* add CompanyUserMailConnectorToMailFacadeBridge

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

daniel-rose committed:
🪲 Fix typo in replace section of composer.json (#550)

julianhyatt committed:
:sparkles: adds jellyfish sales order payone gift card connector (#549)

julianhyatt committed:
:recycle: Refactoring and :white_check_mark: add unit tests (#551)

julianhyatt committed:
:sparkles: Add PayoneCreditMemoGiftCardConnector Bundle (#552)

julianhyatt committed:
:bookmark: prepare pre releases for gift card stuff (#554)

julianhyatt committed:
fix dandelion.json (#555)

julianhyatt committed:
:sparkles: add find method for finding credit memo gift cards (#556)

julianhyatt committed:
:sparkles: Add gift card credit memo bundle (#553)

:sparkles: Add GiftCardCreditMemo bundle
:sparkles: Add OMS condition plugin to check if refund must refund on used gift cards
:sparkles: Add OMS command plugin for refund (dummy)

julianhyatt committed:
alpha release (#557)

julianhyatt committed:
fix payone-credit-memo-gift-card-connector (#558)

julianhyatt committed:
:sparkles: implements proportional coupon value refund on gift card balances (#559)

:sparkles: implements proportional coupon value refund on gift card balances
:recycle: remove unused classes
:recycle: refactoring
:pushpin: adjust dependencies
:bookmark: prepare next alpha release

SpiderDan98 committed:
Fix jellyfish gift card locale (#560)

julianhyatt committed:
:sparkles: refund expenses payone and gift card (#561)

:sparkles: refund expenses to payone
:sparkles: refund proportional gift card amount back on giftcard

daniel-rose committed:
📦 Initialize company unit address api bundle (#562)

- add contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- ...

daniel-rose committed:
📦 Implement contracts of company unit address api bundle (#563)

- provide CRUD over ZED Rest Api
- add transfer definition
- add unit tests
- add entry in dandelion.json for splitting
- ...

julianhyatt committed:
:white_check_mark: add tests (#566)

julianhyatt committed:
:white_check_mark: add tests (#567)

daniel-rose committed:
✨ Cleanup and optimize company api bundle (#564)

- add ci pipeline for split repo
- remove unused query container
- add required transfer definition
- add phpstan config
- ...

daniel-rose committed:
✨ Cleanup and optimize company unit address api bundle (#568)

- add .gitignore
- add phpstan fixes
- cleanup phpstan.neon
- use correct dirs in ci/cd
- excelude orm and transfer classes from code coverage

daniel-rose committed:
✨ Cleanup and optimize company business unit api bundle (#565)

- add ci pipeline for split repo
- remove unused query container
- add required transfer definition
- add phpstan config
- exclude orm and transfer classes from code coverage
- ...

daniel-rose committed:
📦 Initialize company user api bundle (#569)

- add contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- ...

daniel-rose committed:
📦 Implement contracts of company user api bundle (#570)

- provide CRUD over ZED Rest Api
- add unit tests
- add entry to dandelion.json for splitting
- ...

DerCalli committed:
✨ Move concrete product api bundle from fond-of-spryker to fond-of-oryx (#485)

use new namespace (FondOfOryx)
add ci pipeline for split repo
remove unused query container
add required transfer definition
add phpstan config
exclude orm and transfer classes from code coverage
add better unit testing
...

daniel-rose committed:
📦 Initialize product default category assigner bundle (#571)

- add contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- add transfer definition
- ...

daniel-rose committed:
📦 Implement contracts of product default category assigner bundle (#572)

- allows for adding created or updated products to a predefined category
- add unit tests
- add entry to dandelion.json for splitting
- ...

daniel-rose committed:
🪲 Remove unique key from nullable column (#573)

gengjozsef committed:
implement Mail Sending after Customer Registration (#575)

* implement interfaces and initial files

* phpcs fix

* add CompanyUserMailConnectorToMailFacadeBridge

* add Plugin and Business Implementtions

* add implementation for the plugin

* add autoloads in composer.json

* Twig fixes

* phpcs fix

* code cleanup

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

Commit:
Catch all Guzzle exceptions (#576)

* Catch all Guzzle exceptions

* Fix import

gengjozsef committed:
Implement Unit Tests for CompanyUserMailConnector (#577)

* implement interfaces and initial files

* phpcs fix

* add CompanyUserMailConnectorToMailFacadeBridge

* add Plugin and Business Implementtions

* add implementation for the plugin

* add autoloads in composer.json

* Twig fixes

* phpcs fix

* code cleanup

* implement Unit Tests

* phpcs fixes

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
add new entries in dandelion (#578)

* add unit tests in codeception.yml

* add versions for the modules in dandelion.json

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
add  versions for the dependency modul company-users-rest-api-extension in the modules (#579)

* add version for the dependency fond-of-oryx/company-users-rest-api-extension

* increase version for the modules company-user-company-role-connector and company-user-mail-connector

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

daniel-rose committed:
📦 Add cart search rest api extension bundle (#580)

- add zed plugin contracts
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- ...

julianhyatt committed:
:package: introduce gift card proportional value (#574)

:recycle: extracted from JellyfishSalesOrderPayoneGiftCardConnector bundle
:bookmark: add release tag
:recycle: remove extracted code and use facade of new package

daniel-rose committed:
📦 Initialize cart search rest api bundle (#581)

- add zed contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- add transfer definition
- ...

daniel-rose committed:
📦 Implement zed contracts of cart search rest api bundle (#582)

- add unit tests
- implement zed contracts
- provide method to find quotes by QuoteListTransfer
- ...

daniel-rose committed:
✨ Provide zed logic to find quotes to other layer (#583)

- add client to cart search rest api bundle
- add gateway controller for find quotes over client
- add unit tests
- prepare autoload sections in main composer.json
- ...

julianhyatt committed:
:arrow_up: upgrade dependency to :white_check_mark: fix to pass ci (#586)

:arrow_up: upgrade dependency to :white_check_mark: fix to pass ci
:bookmark: prepare alpha releas

julianhyatt committed:
Feature/proportional value mapping payone (#587)

:sparkles: add payone payment method validation
:bookmark: preapre new alpha release

julianhyatt committed:
Feature/jellyfish sales order no payment gift card connector (#588)

:tada: Introduce JellyfishSalesOrderNoPaymentGiftCardConnector
:sparkles: persists the proportional gift card value for orders only paid by gift card
:sparkles: exports the proportional gift card value for orders only paid by gift card
:bookmark: prepare alpha release

julianhyatt committed:
Feature/no payment credit memo (#589)

:tada: Introduce NoPaymentCreditMemo package

julianhyatt committed:
Refactoring/no payment credit memo (#590)

:recycle: Some refactoring
- use data from config
- update proportional instead of creating since this was moved to export plugin in another package
:bookmark: prepare alpha release

julianhyatt committed:
Feature/total export if gift card available (#591)

:sparkles: use totals send from erp for export credit memo if refund has gift cards
:construction_worker: add ci stuff
:white_check_mark: fix to pass ci
:bookmark: prepare release

Commit:
Initialise payone secure invoice package (#584)

* Initialise package

* Add source directory

* Update payone branch

* Add secure invoice credentials mapper (#585)

* Add secure invoice credentials mapper

* Resolve phpcs errors

* Resolve phpcs issues

* Add empty line

* Resolve facade test

* Remove dev config

* Add payone secure invoice bundle

* Update plugin interface

* Fix ci errors

* Fix phpstan and change structure

* Remove unused file

* Remove payone-secure-invoice

* Change mapper structure by mapper types

* Fix phpcs

* Move credentials check to credentials mapper

* Fix phpcs

* Inject credentials into mapper function

julianhyatt committed:
Feature/credit memo improve logging (#593)

* :chart_with_upwards_trend: improve logging if credit memo could not been saved to db
* :bookmark: prepare release

daniel-rose committed:
📦 Add payment cart connector bundle (#594)

- add plugin for deleting payments after cart was saved
- add plugin for deleting payment after cart was saved (backwards compatibly)
- add initial files like composer.json, codeception.yml, ...
- prepare composer.json (main)
- add entry to dandelion.json for creating split repo
- add github actions for split repo
- ...

Commit:
Update payone dependency (#595)

Co-authored-by: Daniel Rose <daniel-rose@gmx.de>

Commit:
Add transactionId mapper (#592)

* Add transactionId mapper

* Add payone secure invoice

* Inject config object instead of value array

* Fix phpcs

* Change config type hint

* Remove unused class

Co-authored-by: Daniel Rose <daniel-rose@gmx.de>

Commit:
Fix phpstan for test directory (#596)

Commit:
Update payone dependency (#598)

julianhyatt committed:
Refactor jellyfish credit memo (#248)

* :recycle: Refactoring
- added tests
- removed unused code
* :bookmark: prepare for release

julianhyatt committed:
Add logging to availability alert cross engage  (#249)

:recycle: refactoring
:construction_worker: add ci stuff
:bookmark: prepare for release

Commit:
Use hash generator for key generation (#600)

* Use hash generator for key generation

* Add parameter

* Add multiline comma

* Add tests for hash generator

* Remove subclearing type check

* Fix phpcs

daniel-rose committed:
🤩 Skip existing packagist entries by creating (#599)

julianhyatt committed:
Bugfix/gift card balance refund (#601)

* :bug: fix refund for orders with 2 same items bought

daniel-rose committed:
Add glue contracts for cart search rest api module (#602)

- prepare transfer definition
- add autoload section in main composer
- ...

daniel-rose committed:
✨ Implement glue contracts for cart search rest api bundle (#603)

- provide endpoint for searching carts
- allow full-text-search
- add sorting
- ...

daniel-rose committed:
✨ Add unit tests for glue layer of cart search rest api bundle (#604)

Commit:
Rename payment product restriction (#597)

Commit:
Remove incorrect comma (#605)

Commit:
Fix replacement product payment restriction (#607)

daniel-rose committed:
🐛 Add some bugfixes for cart search rest api bundle (#606)

- return query object in dependency provider
- allow only configured sort fields
- remove copy paste errors  (e.g. PriceListConstants -> CartSearchRestApiConstants, ...)
- add required pagination by repository class

julianhyatt committed:
Bugfix/gift card refund export (#610)

* :bug: set correct state for refund export
- fix IsExportedConditionPlugin

Commit:
Set trbo session correctly (#609)

* Set trbo session correctly

* Remove unused file

* Fix tests

* Update trbo version

Commit:
Add persistence layer (#611)

* Add persistence layer

* Fix phpcs

* Fix phpstan

daniel-rose committed:
✨ Expand minimalist quotes (#612)

- add bridge to quote facade for using core logic to expand minimalist quotes
- prepare transfer definition
- select "id_quote" for minimalist quotes only
- ...

gengjozsef committed:
Add missing replaces in composer.json (#608)

gengjozsef committed:
add customer-product-list-api (#613)

* add customer-product-list-api

* fix README.md

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

daniel-rose committed:
✨ Allow full-text-search in cart search rest api bundle (#614)

- implement full-text-search in persistence layer
- use more constants
- ...

gengjozsef committed:
fix wrong path for the customer-product-list-api (#615)

* fix path for the customer-product-list-api

* phpcs fixes

* phpcs fix

* phpstan fixes

* php c fixe Throw Exception

* phpcs fix

* phpstan fixes

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

Commit:
Add EventDispatcherPlugin to add cookie (#616)

* Add EventDispatcherPlugin to add cookie

* Fix variable names

* Upgrade trbo

* Fix phpcs

* Change doc block

gengjozsef committed:
implement Unit Tests for customer-product-list-api (#617)

* add Unit Tests

* phpcs fixes

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

Commit:
Remove dev dependency (#618)

Commit:
Update 35up payment connector (#619)

julianhyatt committed:
:tada: Add Gift Card Proportional Value Extension Package (#621)

* :tada: Add Gift Card Proportional Value Extension Package

Commit:
Reduce log noise (#622)

julianhyatt committed:
wip (#623)

daniel-rose committed:
📦 Initialize company business unit cart search rest api bundle (#624)

- add contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- add required transfer definition
- ...

daniel-rose committed:
✨ Add filter fields expander plugin (#626)

- put plugin contract to cart search rest api extension bundle
- use plugin after filter fields are mapped from rest request
- add unit tests
- add autoload stuff
- ...

julianhyatt committed:
Refactoring/gift card proportional value (#627)

:recycle: refactoring
:sparkles: add oms support for creating proportional values within oms run
:sparkles: add possibility to extend via external plugin

daniel-rose committed:
📦 Implement contracts of company business unit cart search rest api bundle (#625)

- implement contracts / interfaces
- add unit tests
- add permission plugin
- ...

julianhyatt committed:
:white_check_mark: add unit tests (#629)

:memo: add README.md

julianhyatt committed:
Release/gift card proportional value (#630)

:bookmark: prepare release

daniel-rose committed:
✨ Add filter fields expander plugins (#631)

- expand filter fields for customer id and company business unit uuid
- ...

gengjozsef committed:
add product-list-api (#628)

* add Interfaces and Initial Files

* phpcs fixes

* add QueryContainer Bridges

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

julianhyatt committed:
Feature/no payment proportional value (#632)

:tada: add GiftCardProportionalValueNoPaymentConnector package
:construction_worker: add ci stuff
:memo: add README.md

gengjozsef committed:
add product list implementation (#633)

* add Interfaces and Initial Files

* phpcs fixes

* add QueryContainer Bridges

* add implementation and unit tests

* phpcs fixes

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
add customer-product-list-api & product-list-api in dandelion (#634)

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
fix composer name (#635)

* fix fix composer name

* add fond-of-oryx/customer-product-list-connector as dependency

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

daniel-rose committed:
🐛 Add some fixes to (x) cart search rest api bundle (#636)

- filter carts which owned by anonymized customer
- use correct join order
- use correct classes
- prepare unit tests
- ...

julianhyatt committed:
Unit tests/proportional value no payment connector (#637)

:white_check_mark: add unit tests
:bookmark: prepare for release

daniel-rose committed:
✨ Expand quote resource relationship by plugin (#638)

- allow items include for endpoint cart-search
- ...

julianhyatt committed:
Feature/proportional value payone connector (#640)

:tada: add GiftCardProportionalValuePayoneConnector bundle

Commit:
Fix psr4 namespace (#639)

* Fix psr4 namespace

* Change namespace

daniel-rose committed:
🚀 Release cart search rest api bundle (#641)

- add required dependencies
- fix local ci problems
- ...

daniel-rose committed:
🚀 Release company business unit cart search rest api bundle (#642)

- add required dependencies
- fix local ci problems
- ...

julianhyatt committed:
:white_check_mark: add unit tests (#643)

:memo: update README.md

gengjozsef committed:
add company-product-list-api (#644)

* add initial files and Interfaces

* phpcs fixes

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

julianhyatt committed:
add missing replace (#645)

gengjozsef committed:
add Facade, Dependency, Plugin, etc implementations and unit tests (#646)

* add initial files and Interfaces

* phpcs fixes

* add Facade, Dependency, Plugin, etc implementations and unit tests

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>
Co-authored-by: Daniel Rose <daniel-rose@gmx.de>

gengjozsef committed:
prepare for release company-product-list-api (#647)

* add replace and version for release

* fix

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

daniel-rose committed:
🐛 Set payload (QuoteListTransfer) to rest resource (#648)

julianhyatt committed:
Feature/jellyfish proportional value mapper (#650)

* :tada: adds JellyfishSalesOrderGiftCardProportionalValueConnector bundle

daniel-rose committed:
✨ Remove plugin for including cart items (to greedy) (#651)

daniel-rose committed:
✨ Add backward compatibility to unit tests (#653)

julianhyatt committed:
:white_check_mark: add unit tests (#652)

* :white_check_mark: add unit tests
* :sparkles: add proportional value mapping to item and expense

daniel-rose committed:
📦 Add fallback product list bundle (#654)

- allow to add fallback blacklist ids and whitelist ids to customer (per plugin)
- make fallback blacklist ids and whitelist ids configurable
- ...

daniel-rose committed:
🐛 Start with correct version (#655)

julianhyatt committed:
Refactore/deprecate modules (#656)

* :boom: deprecate package
* :heavy_minus_sign: remove dependency
* :heavy_plus_sign: add dependency

Commit:
Add secure invoice transaction status validation (#657)

* Add secure invoice transaction status validation

* Fix phpcs

* Change doc type

* Fix phpcs

julianhyatt committed:
Feature/oms credit memo gift card connector (#659)

:art: Copied from GiftCardCreditMemo and renamed to OmsCreditMemoGiftCardConnector
:fire: remove GiftCardCreditMemo bundle
:memo: update readme
:bookmark: prepare release

Commit:
Upgrade payone secure invoice (#658)

* Upgrade payone secure invoice

* Change version number

* Fix ci

* Fix tests

Commit:
Add logic to save and expand product (#620)

* Add logic to save and expand product
* update version of bundles/product-payment-restriction to 1.0.0-beta.2

Co-authored-by: Pascal Fischer <p-fischer@gmx.net>

DerCalli committed:
🧩 product payment restriction (#660)

* add missing dependencies to composer.json

Co-authored-by: Chris <christopher.egger@fondof.de>
Co-authored-by: Pascal Fischert <p-fischer@gmx.ne>

DerCalli committed:
🧩 Feature: product payment restriction (#661)

* for b2c context we need to use another PluginInterface for the create-plugin. for future update there is already a plugin which use the PostCreatePluginInterface.

Co-authored-by: Chris <christopher.egger@fondof.de>
Co-authored-by: Pascal Fischert <p-fischer@gmx.ne>

daniel-rose committed:
✨ Split main workflow steps (#663)

* ✨ Split main workflow steps
- use own steps for ci jobs (phpcs, phpstan, ...)
- increase parallel param for phpcs
- ...

daniel-rose committed:
Release api auth bundle (#664)

gengjozsef committed:
add delivery note number (#662)

* add delivery note number in propel and transfer

* add delivery-note-number in ElasticSearch

* add DeliveryNoteNumber in transfer

* fix Unit Test

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
update version for modules in dandelion.json (#665)

* add delivery note number in propel and transfer

* add delivery-note-number in ElasticSearch

* add DeliveryNoteNumber in transfer

* fix Unit Test

* increase version for erp-delivery-note-page-search and erp-delivery-note

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

daniel-rose committed:
🐛 Use correct path to api auth bundle (#666)

daniel-rose committed:
📦 Add glue application acl bundle (#668)

- manipulate resource protection by plugin
- protect all resource by default
- set resources unprotected by default
- ...

julianhyatt committed:
Feature/map prop value to item (#669)

:sparkles: add plugin for mapping proportional gift card values on sales order items
:white_check_mark: add unit tests
:bookmark: prepare release

daniel-rose committed:
🚀 Release glue application acl (#670)

daniel-rose committed:
📦 Initialize company unit address sales connector bundle (#672)

- add contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- add required transfer definition
- ...

daniel-rose committed:
📦 Implement contracts of company unit address sales connector bundle (#673)

- create relations between sales order address and company unit address
- implement contracts / interfaces
- add unit tests
- add order post save plugin
- ...

daniel-rose committed:
📦 Initialize customer address sales connector bundle (#674)

- add contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- add required transfer definition
- ...

daniel-rose committed:
📦 Implement contracts of customer address sales connector bundle (#675)

- create relations between sales order address and customer address
- implement contracts / interfaces
- add unit tests
- add order post save plugin
- ...

daniel-rose committed:
🐛 Type strict formArray <-> toArray (#676)

- remove company from company unit address transfer
- ...

daniel-rose committed:
🥂 Release x address sales connector bundles (#677)

julianhyatt committed:
Bugfix/fix test (#678)

:white_check_mark: fix test
:pushpin: add temporary "spryker/propel-orm": "<1.16.0" to pass ci

SpiderDan98 committed:
✨ Add credit memo payone debit connector (#671)

DerCalli committed:
Feature/product payment restriction (#667)

* refactor, use entity based filter instead of product-attributes

Co-authored-by: Chris <christopher.egger@fondof.de>
Co-authored-by: Pascal Fischert <p-fischer@gmx.ne>

SpiderDan98 committed:
✨Feature: add debit is refunded condition plugin (#680)

SpiderDan98 committed:
✨ Feature: add is debited condition plugin (#681)

gengjozsef committed:
add erp order item position (#683)

* add position in erp_order_schema.xml

* add position in transfer objects

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
Sort Erp Order Items based on position in case of the ElasticSearch export (#684)

* sort Items based on position

* phpcs fix

* phpstan fixes

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
release implementation regarding erp order item position information (#685)

* sort Items based on position

* phpcs fix

* phpstan fixes

* increase versions for erp-order and erp-order-page-search

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

gengjozsef committed:
increase version for erp-order-page-search-rest-api (#686)

* sort Items based on position

* phpcs fix

* phpstan fixes

* increase versions for erp-order and erp-order-page-search

* increase version for erp-order-page-search-rest-api

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

SpiderDan98 committed:
✨ Feature: add partial debit (#687)

SpiderDan98 committed:
⬆️ Increase: version for credit memo packages (#688)

SpiderDan98 committed:
🐛 Fix: payone debit connector composer json (#689)

Commit:
Remove hidden character (#690)

SpiderDan98 committed:
🧹 Chore: add allow plugin codesniffer (#691)

DerCalli committed:
🧩 Feature: get blacklisted country-list (#693)

* new client method to get blacklisted countries as array
* update product-country-restriction-checkout-connector to 1.1.0

DerCalli committed:
🧩 Feature: product country restriction (#694)

* update product-country-restriction-checkout-connector to 1.1.1
* clear billing- and shipping-address if quote is invalid

daniel-rose committed:
✨ Allow newest version of "spryker/propel-orm" (#696)

- prepare unit tests
- add phpstan fixes
- use mock for object collection
- ...

daniel-rose committed:
✨ Add more logging to splittable checkout bundle (#695)

daniel-rose committed:
📦 Add splittable quote nopayment connector bundle (#697)

- change payment info if price to pay equals zero (splitted quote expander plugin)
- add unit tests
- add autoload stuff to main composer.json
- prepare dandelion.json
- ...

daniel-rose committed:
🤙🏽 Add better recalculation process for splittable quotes (#698)

DerCalli committed:
🧩 product country restriction checkout (#700)

* unit tests
* api specification
* Use better naming
* refactor, use new collection-transfer for client response

Co-authored-by: Pascal Fischert <p-fischer@gmx.ne>
Co-authored-by: Daniel Rose <daniel-rose@gmx.de>

daniel-rose committed:
🫗 Add company role name filter (#701)

- allow to filter company user search result by company role names
- filter param "company-role-name[]" from "company-user-search"-endpoint
- add missing unit tests
- remove specified classes from code coverage
- ...

DerCalli committed:
🧩 reset address on product country restriction (#702)

* add plugin to reset customer-address-data on restricted item
* unit-test for new plugin
* update product-country-restriction-checkout-connector to 1.1.1

Co-authored-by: Pascal Fischert <p-fischer@gmx.ne>

julianhyatt committed:
prepare release (#704)

daniel-rose committed:
📦 Initialize vertigo price product price list bundle (#705)

- add contracts / interfaces
- add initial files like composer.json, codeception.yml, ...
- add prepare composer.json (main)
- add github actions for split repo
- add required transfer definition
- ...

daniel-rose committed:
📦 Implement contracts of vertigo price product price list bundle (#706)

- allow to request missing prices by vertigo ms
- implement contracts / interfaces
- add unit tests
- add console command
- ...

daniel-rose committed:
🤑 Allow to request price list prices per sku (#707)

- add console command to trigger request
- add facade method to provide the functionality
- add unit tests
- ...

gengjozsef committed:
Add base structure for the company-user-archive implementation (#709)

* add initial implementation for the company-user-archive

* add propel schema

* add Interfaces

* fix method name in the Facade interface

* code cleanup

* phpstan fix

Co-authored-by: Jozsef Geng <jozsef.geng@fondof.de>

daniel-rose committed:
🤑 Allow to request price list prices per product concrete (#708)

- add plugins to trigger request
- add facade method to provide the functionality
- add unit tests
- ...

daniel-rose committed:
🐛 Add required dependencies (#711)

gengjozsef committed:
:dna: Implement company-user-archive (#712)

* add implementation for the interfaces

* phpcs fixes

* code cleanup

* phpcs fixes

* phpstan fixes

* add Plugin and Facade Test

daniel-rose committed:
🐛 Encode body before sending request (#713)

daniel-rose committed:
🎢 Release bundles which use the newest log bundle (#714)

gengjozsef committed:
:abacus: add Unit Test for company-user-archive  (#715)

* add implementation for the interfaces

* phpcs fixes

* code cleanup

* phpcs fixes

* phpstan fixes

* add Plugin and Facade Test

* add BusinessFactory and Model Test

* phpcs fixes

* phpcs fixes

* add bundles/company-user-archive entry in codeception.yml

* phpcs fix

* code cleanup

* phpcs fix

DerCalli committed:
🐌 TRBO A/B testing (#716)

* testing without session


Co-authored-by: Pascal Fischert <p-fischer@gmx.ne>

gengjozsef committed:
prepare company-user-archive to release (#717)

gengjozsef committed:
add missing company-user-archive in replace section in the composer.json (#719)

gengjozsef committed:
Merge branch 'main' of https://github.com/fond-of-oryx/fond-of-oryx into main

# Conflicts:
#	bundles/company-role-search-rest-api/src/FondOfOryx/Zed/CompanyRoleSearchRestApi/CompanyRoleSearchRestApiDependencyProvider.php
#	bundles/company-role-search-rest-api/src/FondOfOryx/Zed/CompanyRoleSearchRestApi/Persistence/CompanyRoleSearchRestApiPersistenceFactory.php
#	bundles/company-role-search-rest-api/src/FondOfOryx/Zed/CompanyRoleSearchRestApi/Persistence/CompanyRoleSearchRestApiRepository.php
#	bundles/company-role-search-rest-api/src/FondOfOryx/Zed/CompanyRoleSearchRestApi/Persistence/CompanyRoleSearchRestApiRepositoryInterface.php
#	bundles/company-user-mail-connector/src/FondOfOryx/Zed/CompanyUserMailConnector/Business/Model/Mail/MailHandler.php
#	bundles/company-user-search-rest-api/src/FondOfOryx/Zed/CompanyUserSearchRestApi/Persistence/CompanyUserSearchRestApiRepository.php
#	bundles/erp-order-page-search-rest-api/src/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapper.php
#	bundles/erp-order-page-search-rest-api/src/FondOfOryx/Shared/ErpOrderPageSearchRestApi/Transfer/erp_order_page_search_rest_api.transfer.xml
#	bundles/erp-order-page-search-rest-api/tests/FondOfOryx/Glue/ErpOrderPageSearchRestApi/Model/Mapper/ErpOrderMapperTest.php
#	bundles/erp-order-page-search/composer.json
#	bundles/erp-order-page-search/src/FondOfOryx/Zed/ErpOrderPageSearch/Business/Mapper/ErpOrderPageSearchDataMapper.php
#	bundles/erp-order-page-search/src/FondOfOryx/Zed/ErpOrderPageSearch/Business/Publisher/ErpOrderPageSearchPublisher.php
#	bundles/erp-order-page-search/tests/FondOfOryx/Zed/ErpOrderPageSearch/Business/Mapper/ErpOrderPageSearchDataMapperTest.php
#	bundles/erp-order/composer.json
#	bundles/erp-order/src/FondOfOryx/Shared/ErpOrder/Transfer/erp_order.transfer.xml
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Business/ErpOrderBusinessFactory.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Business/ErpOrderFacade.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Business/ErpOrderFacadeInterface.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/ErpOrderDependencyProvider.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/ErpOrderEntityManager.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/ErpOrderEntityManagerInterface.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/ErpOrderPersistenceFactory.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/ErpOrderRepository.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/ErpOrderRepositoryInterface.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/Propel/Mapper/EntityToTransferMapper.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/Propel/Mapper/EntityToTransferMapperInterface.php
#	bundles/erp-order/src/FondOfOryx/Zed/ErpOrder/Persistence/Propel/Schema/erp_order.schema.xml
#	bundles/erp-order/tests/FondOfOryx/Zed/ErpOrder/Business/ErpOrderFacadeTest.php
#	bundles/jellyfish-sales-order-company/composer.json
#	bundles/jellyfish-sales-order-company/src/FondOfOryx/Shared/JellyfishSalesOrderCompany/Transfer/jellyfish_sales_order_company.transfer.xml
#	bundles/jellyfish-sales-order-company/src/FondOfOryx/Zed/JellyfishSalesOrderCompany/Business/Expander/JellyfishOrderExpander.php
#	bundles/jellyfish-sales-order-company/src/FondOfOryx/Zed/JellyfishSalesOrderCompany/Business/JellyfishSalesOrderCompanyBusinessFactory.php
#	bundles/jellyfish-sales-order-company/src/FondOfOryx/Zed/JellyfishSalesOrderCompany/JellyfishSalesOrderCompanyDependencyProvider.php
#	bundles/jellyfish-sales-order-company/tests/FondOfOryx/Zed/JellyfishSalesOrderCompany/Business/Expander/JellyfishOrderExpanderTest.php
#	bundles/jellyfish-sales-order-company/tests/FondOfOryx/Zed/JellyfishSalesOrderCompany/Business/JellyfishSalesOrderCompanyBusinessFactoryTest.php
#	bundles/shipment-table-rate-data-import/src/FondOfOryx/Zed/ShipmentTableRateDataImport/ShipmentTableRateDataImportConfig.php
#	bundles/shipment-table-rate-data-import/src/FondOfOryx/Zed/ShipmentTableRateDataImport/ShipmentTableRateDataImportDependencyProvider.php
#	bundles/shipment-table-rate/src/FondOfOryx/Shared/ShipmentTableRate/Transfer/shipment_table_rate.transfer.xml
#	bundles/splittable-checkout-rest-api/src/FondOfOryx/Glue/SplittableCheckoutRestApi/Processor/Builder/SplittableCheckoutRestResponseBuilder.php
#	bundles/splittable-checkout-rest-api/src/FondOfOryx/Glue/SplittableCheckoutRestApi/Processor/Builder/SplittableCheckoutRestResponseBuilderInterface.php
#	bundles/splittable-checkout-rest-api/src/FondOfOryx/Glue/SplittableCheckoutRestApi/Processor/SplittableCheckout/SplittableCheckoutProcessor.php
#	bundles/splittable-checkout-rest-api/src/FondOfOryx/Glue/SplittableCheckoutRestApi/SplittableCheckoutRestApiDependencyProvider.php
#	bundles/splittable-checkout-rest-api/src/FondOfOryx/Glue/SplittableCheckoutRestApi/SplittableCheckoutRestApiFactory.php
#	bundles/splittable-checkout-rest-api/src/FondOfOryx/Shared/SplittableCheckoutRestApi/Transfer/splittable_checkout_rest_api.transfer.xml
#	bundles/splittable-checkout-rest-api/src/FondOfOryx/Zed/SplittableCheckoutRestApi/Business/Processor/PlaceOrderProcessor.php
#	bundles/splittable-checkout-rest-api/tests/FondOfOryx/Glue/SplittableCheckoutRestApi/Processor/Builder/SplittableCheckoutRestResponseBuilderTest.php
#	bundles/splittable-checkout-rest-api/tests/FondOfOryx/Glue/SplittableCheckoutRestApi/Processor/SplittableCheckout/SplittableCheckoutProcessorTest.php
#	bundles/splittable-checkout-rest-api/tests/FondOfOryx/Glue/SplittableCheckoutRestApi/SplittableCheckoutRestApiFactoryTest.php
#	bundles/splittable-checkout/src/FondOfOryx/Shared/SplittableCheckout/Transfer/splittable_checkout.transfer.xml
#	bundles/splittable-checkout/src/FondOfOryx/Zed/SplittableCheckout/Business/Workflow/SplittableCheckoutWorkflow.php
#	bundles/splittable-checkout/tests/FondOfOryx/Zed/SplittableCheckout/Business/Workflow/SplittableCheckoutWorkflowTest.php
#	composer.json
#	dandelion.json

gengjozsef committed:
Merge branch 'main' of https://github.com/fond-of-oryx/fond-of-oryx into main

gengjozsef committed:
Merge branch 'main' of https://github.com/fond-of-oryx/fond-of-oryx into main

gengjozsef committed:
Impplement Customer Filter Zed Classes

gengjozsef committed:
Impplement Customer Filter Zed Classes

gengjozsef committed:
code cleanup

gengjozsef committed:
phpcs fixes

gengjozsef committed:
code cleanup and phpcs fixes

gengjozsef committed:
phpcs fixes

gengjozsef committed:
phpstan fixes

gengjozsef committed:
fix Unit test

gengjozsef committed:
phpcs fixe

gengjozsef committed:
unit test fix and code cleanup

gengjozsef committed:
revert codeception.yml

gengjozsef committed:
phpcs fix

gengjozsef committed:
code cleanup

gengjozsef committed:
Merge branch 'main' of https://github.com/fond-of-oryx/fond-of-oryx into main

gengjozsef committed:
Merge branch 'main' into feature/product-list-search-rest-api-customer-filter

gengjozsef committed:
Implement filter in Glue und Client

gengjozsef committed:
phpcs fix

gengjozsef committed:
phpstan fixes

gengjozsef committed:
unit test Fix

gengjozsef committed:
revert

gengjozsef committed:
revert codeception.yml

gengjozsef committed:
extend Unit Tests

gengjozsef committed:
extend Unit Tests
